lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -429,10 +429,20 @@ struct ExprVisitor {
   }
 
   Value visit(const slang::ast::MemberAccessExpression &expr) {
-    return builder.create<moore::StructExtractOp>(
-        loc, context.convertType(*expr.type),
-        builder.getStringAttr(expr.member.name),
-        context.convertExpression(expr.value()));
+    auto type = expr.value().type;
+    if (type->isStruct()) {
+      return builder.create<moore::StructExtractOp>(
+          loc, context.convertType(*expr.type),
+          builder.getStringAttr(expr.member.name),
+          context.convertExpression(expr.value()));
+    }
+    if (type->isPackedUnion() || type->isUnpackedUnion()) {
+      return builder.create<moore::UnionExtractOp>(
+          loc, context.convertType(*expr.type),
+          builder.getStringAttr(expr.member.name),
+          context.convertExpression(expr.value()));
+    }
```

Comment:
`context.convertExpression()` may trigger `segmentation fault` if it returns `{}`, so I think it would be better if there's an `if` branch.

Reply:
Good advice, I will add another check for this type.